### PR TITLE
Switch to GitHub Actions

### PR DESCRIPTION
**Pull request by shaardie**
Static code analysis with flake8 as well as the docs generation and
deployment moved to GitHub Actions.

There is still installation on Buster and alembic migration tests in CircleCI, but I would like to remove them on the long run. So I did not see any reason to move them too.

### CODE REVIEW

**Comment by shaardie** [comment]
Some tests are not properly shown, but you can see here https://github.com/shaardie/emissions-api/tree/switch-to-github-actions that they have run successfully.

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+name: Lint
+
+on: push
```
Comment (lkiesow) [inline .github/workflows/lint.yml:3]: This should probably run for pull requests as well, shouldn't it?

**Review by lkiesow — CHANGES_REQUESTED** [review]
Looks good except that the linting should probably happen on pull requests as well

File: .github/workflows/lint.yml (hunk 1 above)
Comment (shaardie) [reply .github/workflows/lint.yml]: Added `pull_request` for both workflows

**Pull request merged**